xbmc/cores/AudioEngine/Engines/ActiveAE/ActiveAE.cpp

```diff
@@ -152,6 +152,13 @@ void CEngineStats::GetDelay(AEDelayStatus& status, CActiveAEStream *stream)
     status.delay += static_cast<double>(m_bufferedSamples) *
                     m_sinkFormat.m_streamInfo.GetDuration() / 1000;
 
+  if (!m_pcmOutput && m_sinkNeedIecPack &&
+      m_sinkFormat.m_streamInfo.m_type == CAEStreamInfo::STREAM_TYPE_TRUEHD)
+  {
+    // take into account MAT packer latency
+    status.delay += 12.0 * m_sinkFormat.m_streamInfo.GetDuration() / 1000;
```

Comment:
This here results in 240 ms. Isn't it rather (12.0 / 24.0), e.g. 1/2 of 20 ms?

Reply:
You're right, I think I've somehow related it to the 250 ms commented on the forum, but it has to be something else.

12 + 12 + 12 -----> MAT packer ----> output 24

latency is 12 units which is 10 ms (half MAT package)


Maybe exact 200 ms is the "WATER_LEVEL" duration of buffer (when is filled with 10 MAT packages or 10 x 24 audio units)

Reply:
Okay, that maybe means we are on average 12 units off, so around:
```
    status.delay += 0.5 * m_sinkFormat.m_streamInfo.GetDuration() / 1000;
```
But only for IEC, cause for RAW we package 24 at once.

Reply:
But actually also only once. Cause after that always 24 units can be output, just with 12 units (10 ms) delay.

Reply:
RAW packer is transparent with no latency

24 units ----->  RAW packer ----> output 24 units in the same function call (no latency)

Reply:
Jep - exactly. That's why it's only relevant for the very first package and only for IEC. Code is correct as you have it right now!

Reply:
Now is:
`status.delay += m_sinkFormat.m_streamInfo.GetDuration() / 1000 / 2;`

should be equivalent although i can change

Reply:
One important thing here is that I cannot reproduce any delay in my setup changing delay here (even 250 ms) but maybe for users that have this issue, can be compensated here (only for TrueHD IEC) and works fine for all. 

As seems not affect to users that NOT have a/v sync issues.